closure/goog/array/array.js

```diff
@@ -496,8 +496,8 @@ goog.array.count = function(arr, f, opt_obj) {
  *     for every element. This function takes 3 arguments (the element, the
  *     index and the array) and should return a boolean.
  * @param {S=} opt_obj An optional "this" context for the function.
- * @return {T|null} The first array element that passes the test, or null if no
- *     element is found.
+ * @return {T|string|null} The first array element that passes the test, or
```

Comment:
Array.prototype.find.call("foo",...) would be the equivalent
On Feb 14, 2016 6:29 PM, "Michael Zhou" <notifications@github.com> wrote:

> In closure/goog/array/array.js
> <https://github.com/google/closure-library/pull/655#discussion_r52854849>:
>
> > @@ -496,8 +496,8 @@ goog.array.count = function(arr, f, opt_obj) {
> >   *     for every element. This function takes 3 arguments (the element, the
> >   *     index and the array) and should return a boolean.
> >   * @param {S=} opt_obj An optional "this" context for the function.
> > - * @return {T|null} The first array element that passes the test, or null if no
> > - *     element is found.
> > + * @return {T|string|null} The first array element that passes the test, or
>
> https://github.com/google/closure-compiler/blob/master/externs/es6.js#L1424
>
> I don't think "foo".find works.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/closure-library/pull/655/files#r52854849>.
>